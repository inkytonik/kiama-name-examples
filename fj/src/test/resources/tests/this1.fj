class A extends Object {
  A() {
    super();
  }
  A self() {
    return this.this;
  }
}
new A().self()
