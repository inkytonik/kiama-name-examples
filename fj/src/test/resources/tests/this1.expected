src/test/resources/tests/this1.fj:6:12:info: this.this: unknown
    return this.this;
           ^
src/test/resources/tests/this1.fj:6:12:info: this: A
    return this.this;
           ^
src/test/resources/tests/this1.fj:9:1:info: new A().se ...: A
new A().self()
^
src/test/resources/tests/this1.fj:9:1:info: new A(): A
new A().self()
^
A
